Sources/SwiftDocC/Infrastructure/Link Resolution/PathHierarchy.swift

```diff
@@ -157,9 +157,18 @@ struct PathHierarchy {
                 if let targetNode = nodes[relationship.target], targetNode.name == expectedContainerName {
                     targetNode.add(symbolChild: sourceNode)
                     topLevelCandidates.removeValue(forKey: relationship.source)
-                } else if let targetNodes = allNodes[relationship.target] {
-                    for targetNode in targetNodes where targetNode.name == expectedContainerName {
+                } else if var targetNodes = allNodes[relationship.target] {
+                    // Only match nodes that have the expected name
+                    targetNodes.removeAll(where: { $0.name != expectedContainerName })
+                    
+                    // If the source was added in an extension symbol graph file, then its target node won't be found in the same symbol graph file (in `nodes`).
```

Comment:
Not sure what this comment is trying to explain.

Reply:
It describes the reason why execution would end up in this else-if branch

Reply:
Oh ok I get what is being described now, might be worth moving it up to where the else-if is then?

Reply:
I moved the comment up and added some more information to it.